src/common/src/utils.rs

```diff
@@ -12,3 +12,35 @@ pub fn init_nones<T, A: FixedSizeArray<Option<T>>>() -> A {
     res
 }
 
+/// Allocates a global array of static size, initially set to zero. When this
+/// macro is hit, it will initialize the array to the value given and return a
+/// `&'static mut` reference to it.
+///
+/// Note that you will have to specify the array-size as an argument, but a
+/// wrong size will result in a compile-time error. This argument will be
+/// removed if `std::mem::size_of<T>` ever becomes a `const` function.
+#[macro_export]
+macro_rules! static_init {
+    ($e:expr, $T:ty, $size:expr) => {
```

Comment:
Stylistically I think it would be nicer to have the type precede the expression, so that the caller would write something like:

```rust
let gpio = static_init!([GPIOPin; 43],
    [ PA[0],
      PA[1],
    ... ]);
```

That way it's easier to read what the type of the variable is.